src/main/window.ts

```diff
@@ -12,4 +13,21 @@ export function registerWindowHandlers() {
     }
     mainWindow.focus()
   })
+
+  ipcMain.handle('openDialog', async () => {
+    const result = await dialog.showOpenDialog({ properties: ['openDirectory'] })
+    if (result.canceled) throw new Error('Dialog cancelled')
+    return result.filePaths
+  })
+
+  ipcMain.handle('saveFile', (_event, path: string, content: string) => {
+    const buff = Buffer.from(content, 'utf-8')
+    try {
+      writeFile(path, buff, err => {
+        if (err) throw err
+      })
+    } catch (err) {
+      throw new Error()
+    }
+  })
```

Comment:
Use writeFile from "fs/promises" instead. It would be bette since you can wait the file to be really created.

```suggestion
  ipcMain.handle('saveFile', async (_event, path: string, content: string) => {
    const buff = Buffer.from(content, 'utf-8')
    await writeFile(path, buff)

  })
```